src/Misc/PerformanceViewer/performanceViewerCollector.ts

```diff
@@ -211,6 +219,148 @@ export class PerformanceViewerCollector {
         this.metadataObservable.notifyObservers(this._datasetMeta);
     }
 
+    /**
+     * Completely clear, data, ids, and strategies saved to this performance collector.
+     */
+    public clear() {
+        this.datasets.data = new DynamicFloat32Array(initialArraySize);
+        this.datasets.ids.length = 0;
+        this.datasets.startingIndices = new DynamicFloat32Array(initialArraySize);
+        this._datasetMeta.clear();
+        this._strategies.clear();
+        this._hasLoadedData = false;
+    }
+
+    /**
+     * Accessor which lets the caller know if the performance collector has data loaded from a file or not!
+     * Call clear() to reset this value.
+     * @returns true if the data is loaded from a file, false otherwise.
+     */
+    public get hasLoadedData(): boolean {
+        return this._hasLoadedData;
+    }
+
+    /**
+     * Given a string containing file data, this function parses the file data into the datasets object.
+     * It returns a boolean to indicate if this object was successfully loaded with the data.
+     * @param data string content representing the file data.
+     * @returns true if the data was successfully loaded, false otherwise.
+     */
+    public loadFromFileData(data: string): boolean {
+        const lines =
+            data.split('\n')
+                .map((line) => (
+                    line.replace('\r', '')

```

Comment:
why not `replaceAll` (or replace with global regex) on the entire data? seems like we don't need \r's

Reply:
I use replace with a global regex now! replaceAll didn't seem to be available which is why I hadn't used it.